### PR TITLE
UI Components (Matthew)

### PR DESCRIPTION
**Pull request by matthewferreira99**

### CODE REVIEW

**Comment by matthewferreira99** [comment]
Closed this to make a new pull with updated stuff

**Comment by k-kruusi** [comment]
avoid pushing your ipch files and other files you didnt directly alter, typically you should only have the files you edited and or changes to the project file if you added or removed any files.

File: example/example_main.cpp [hunk 1]
```diff
@@ -14,6 +15,9 @@ int main(int argc, char* argv[])
 	engine->scene_manager->AddScene("example_scene", std::make_unique<ExampleScene>());
 	engine->scene_manager->SwitchScene("example_scene");
 
+	engine->ui_manager->AddUI("example_ui", std::make_unique<ExampleUI>());
```
Comment (k-kruusi) [inline example/example_main.cpp:18]: avoid using hardcoded strings, use a constant instead.

File: example/ui/example_ui.cpp [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+void ExampleUI::BeginScene()
+{
+	auto game_object = GetSECManager()->AddGameObject<pav::IGameObjectBase>("Example_UIObject");
+	game_object->AddComponent<pav::TransformComponent>();
+	game_object->AddComponent<pav::SDLImageComponent>();
+
+	/* This is purely for testing purposes since the Component Manager and Render calss weren't in my version
+		Delete in final build
+	pav::TransformComponent *t;
```
Comment (k-kruusi) [inline example/ui/example_ui.cpp:14]: try not to push commented out code to master. you can keep these changes on your local branch in staging.

**Pull request closed without merging**